packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -419,6 +445,20 @@ export class TaskDefinition extends TaskDefinitionBase {
     }
   }
 
+  private renderInferenceAccelerators(): CfnTaskDefinition.InferenceAcceleratorProperty[] {
+    if (isFargateCompatible(this.compatibility)) {
+      throw new Error('Cannot use inference accelerators on tasks that run on Fargate');
+    }
```

Comment:
hmm why do we validate the compatibility again? isn't this duplicated with https://github.com/aws/aws-cdk/pull/13950/files?short_path=2215a07#diff-8a9e5a6a7ba5a3a49b385e40b376a95f99d2781512c60e392dbf3b9056569cf5R403 ? I think we should probly just keep one of them: either scilently set it to be `undefined` or error out.